gov/politeia/proposals.go

```diff
@@ -90,371 +87,509 @@ func NewProposalsDB(politeiaURL, dbPath string) (*ProposalDB, error) {
+	filterByVoteStatus ...int) ([]*pitypes.ProposalRecord, int, error) {
+	// Sanity check
 	if db == nil || db.dbP == nil {
 		return nil, 0, errDef
 	}
 
 	var query storm.Query
+
 	if len(filterByVoteStatus) > 0 {
-		// Filter by the votes status
 		query = db.dbP.Select(q.Eq("VoteStatus",
-			pitypes.VoteStatusType(filterByVoteStatus[0])))
+			ticketvotev1.VoteStatusT(filterByVoteStatus[0])))
 	} else {
 		query = db.dbP.Select()
 	}
 
 	// Count the proposals based on the query created above.
-	totalCount, err = query.Count(&pitypes.ProposalInfo{})
+	totalCount, err := query.Count(&pitypes.ProposalRecord{})
 	if err != nil {
-		return
+		return nil, 0, err
 	}
 
 	// Return the proposals listing starting with the newest.
+	var proposals []*pitypes.ProposalRecord
 	err = query.Skip(offset).Limit(rowsCount).Reverse().OrderBy("Timestamp").
 		Find(&proposals)
-	if err != nil && err != storm.ErrNotFound {
+	if err != nil && !errors.Is(err, storm.ErrNotFound) {
 		log.Errorf("Failed to fetch data from Proposals DB: %v", err)
-	} else {
-		err = nil
 	}
 
-	return
+	return proposals, totalCount, nil
 }
 
-// ProposalByToken returns the single proposal identified by the provided token.
-func (db *ProposalDB) ProposalByToken(proposalToken string) (*pitypes.ProposalInfo, error) {
+// ProposalByToken retrieves the proposal for the given token argument.
+//
+// Satisfies the PoliteiaBackend interface.
+func (db *ProposalsDB) ProposalByToken(token string) (*pitypes.ProposalRecord, error) {
 	if db == nil || db.dbP == nil {
 		return nil, errDef
 	}
 
-	return db.proposal("TokenVal", proposalToken)
+	return db.proposal("Token", token)
 }
 
-// ProposalByRefID returns the single proposal identified by the provided refID.
-// RefID is generated from the proposal name and used as the descriptive part of
-// the URL to proposal details page on the /proposal page.
-func (db *ProposalDB) ProposalByRefID(RefID string) (*pitypes.ProposalInfo, error) {
-	if db == nil || db.dbP == nil {
-		return nil, errDef
+// fetchVettedTokens fetches all vetted tokens ordered by the timestamp of
+// their last status change.
+func (db *ProposalsDB) fetchVettedTokensInventory() ([]string, error) {
+	page := 0
+	vettedTokens := []string{}
+	for {
+		inventoryReq := recordsv1.InventoryOrdered{
+			State: recordsv1.RecordStateVetted,
+			Page:  uint32(page + 1),
+		}
+		reply, err := db.client.RecordInventoryOrdered(inventoryReq)
+		if err != nil {
+			return nil, fmt.Errorf("Pi client RecordInventoryOrdered err: %v",
+				err)
+		}
+
+		vettedTokens = append(vettedTokens, reply.Tokens...)
+
+		// Break loop if we fetch last page
+		if len(reply.Tokens) < int(recordsv1.InventoryPageSize) {
+			break
+		}
+	}
+	return vettedTokens, nil
+}
+
+// fetchRecordDetails fetches the record details of the given proposal tokens.
+func (db *ProposalsDB) fetchRecordDetails(tokens []string) (map[string]recordsv1.Record, error) {
+	records := make(map[string]recordsv1.Record, len(tokens))
```

Comment:
Why not go with `map[string]*ecordsv1.Record` since `Record` is not a small struct and you already have a pointer from `RecordDetails`.